docs/gallery/general/extensions.py

```diff
@@ -326,8 +326,8 @@ class PotatoSack(MultiContainerInterface):
 from pynwb import NWBHDF5IO, NWBFile
 
 potato_sack = PotatoSack(source='pantry',
-                         potatos=Potato(name='potato1', age=2.3, weight=3.0,
-                                        source='the ground'))
+                         potatoes=Potato(name='potato1', age=2.3, weight=3.0,
+                                         source='the ground'))
```

Comment:
This was also intentional. This is the way you need to write it to get the api to work. We should really file an issue about this.

Reply:
So, the `PotatoSack` class builds its attributes using `__clsconf__` (why not use [`attrs`](https://github.com/python-attrs/attrs)?)

My initial PR changes broke your CI due to the mismatch between the `__clsconf__` attribute and the actual keyword assignment. The latest version of the PR fixed the spelling in both places.

What is the current intentional behaviour? All the tests pass, but is there something not being tested?

Reply:
Yes, I think there is something not being tested. As written now I believe it will create an orphaned dataset, but catching it will require reading in the file created in the demo. That would help for pedagogical completeness as well, so I'll contribute a PR adding that to the demo.